### PR TITLE
OADP-3716: seg fault in openshift-adp-controller-manager

### PR DESCRIPTION
**Pull request by anarnold97**
### JIRA

* [OADP-3716](https://issues.redhat.com/browse/OADP-3716)

### VERSIONS

* OCP 4.12 → branch/enterprise-4.12
* OCP 4.13 → branch/enterprise-4.13
* OCP 4.14 → branch/enterprise-4.14
* OCP 4.15 → branch/enterprise-4.15
* OCP 4.16 → branch/enterprise-4.16

### Link to docs preview:

* [OADP plugins known issues](https://73315--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-features-plugins#oadp-features-plugins-known-issues_oadp-features-plugins)
* [Troubleshooting - OADP plugins known issues](https://73315--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/troubleshooting#oadp-features-plugins-known-issues_oadp-troubleshooting)

### QE review:
- [X ] QE has approved this change. [LGTM](https://github.com/openshift/openshift-docs/pull/73315#pullrequestreview-1955790702)

Additional information:

### CODE REVIEW

File: modules/oadp-features-plugins-known-issues.adoc
Comment (anarnold97) [inline modules/oadp-features-plugins-known-issues.adoc:55]:
```suggestion
== `openshift-adp-controller` segmentation fault
```

File: modules/oadp-features-plugins-known-issues.adoc
Comment (anarnold97) [inline modules/oadp-features-plugins-known-issues.adoc:68]:
```suggestion
=== `openshift-adp-controller` segmentation fault workaround
```

File: modules/oadp-features-plugins-known-issues.adoc
Comment (anarnold97) [inline modules/oadp-features-plugins-known-issues.adoc:70]:
```suggestion
To avoid the `openshift-adp-controller-manager` pod failing with a crash loops segmentation fault, ensure you have either `velero` or `cloudstorage` defined, but not both, when configuring a DPA.
```

File: modules/oadp-features-plugins-known-issues.adoc
Comment (anarnold97) [inline modules/oadp-features-plugins-known-issues.adoc:61]:
```suggestion
* If you have both `velero` and  `cloudstorage` defined, the `openshift-adp-controller-manager` will fail.
```

File: modules/oadp-features-plugins-known-issues.adoc
Comment (anarnold97) [inline modules/oadp-features-plugins-known-issues.adoc:64]:
```suggestion
For more details about this issue, see link:https://issues.redhat.com/browse/OADP-1054[OADP-1054].
```

**Review by weshayutin — APPROVED** [review]
/lgtm

File: modules/oadp-features-plugins-known-issues.adoc
Comment (michaelryanpeter) [inline modules/oadp-features-plugins-known-issues.adoc:55]: The [repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references) prohibit using markup in headings. Also, IBM Style dictates using [sentence case](https://www.ibm.com/docs/en/ibm-style?topic=grammar-capitalization#headings-titles-and-banners) for titles. I recognize that this isn't always possible, but I think that you could easily do one of the following: 
* Use a generic reference for the controller: "OpenShift ADP Controller segmentation fault"
* If the specific object reference is important, rephrase and leave off the backtics (`): "Segmentation fault in/for openshift-adp-controller"

File: modules/oadp-features-plugins-known-issues.adoc
Comment (michaelryanpeter) [inline modules/oadp-features-plugins-known-issues.adoc:57]:
```suggestion
If you configure a DPA with both `cloudstorage` and `restic` enabled, the `openshift-adp-controller-manager` pod crashes and restarts indefinitely until the pod fails with a crash loop segmentation fault.
```
Does this change the meaning to much? If so, I find this a bit easier to understand.

File: modules/oadp-features-plugins-known-issues.adoc
Comment (michaelryanpeter) [inline modules/oadp-features-plugins-known-issues.adoc:59]: I am having a hard time understanding the relationship between Restic, Velero, and CloudStorage. See my second suggestion above. 

Is Restic a part of Velero or CloudStorage? Or is Restic it an independent object fails because both CloudStorage and Velero are configured? Apologies if this distinction would be obvious to customers, but I am getting a little disoriented.

File: modules/oadp-features-plugins-known-issues.adoc
Comment (michaelryanpeter) [reply modules/oadp-features-plugins-known-issues.adoc:57]: Now that I have gotten to the workaround, I wonder if this would be more accurate?
```suggestion
When you a DPA with `cloudstorage` and `restic` enabled, if you specify values for the `velero` API, the `openshift-adp-controller-manager` pod fails with a crash loop segmentation fault.
```

File: modules/oadp-features-plugins-known-issues.adoc
Comment (michaelryanpeter) [inline modules/oadp-features-plugins-known-issues.adoc:62]: IBM Style says to [avoid future tense](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#tense).

```suggestion
You can have either `velero` or `cloudstorage` defined, as they are mutually exclusive fields.

* If you have both `velero` and  `cloudstorage` defined, the `openshift-adp-controller-manager` fails.
* If you have neither `velero` nor `cloudstorage` defined, the `openshift-adp-controller-manager` fails.
```

File: modules/oadp-features-plugins-known-issues.adoc
Comment (michaelryanpeter) [inline modules/oadp-features-plugins-known-issues.adoc:68]: See comment above about heading guidelines.

File: modules/oadp-features-plugins-known-issues.adoc
Comment (michaelryanpeter) [inline modules/oadp-features-plugins-known-issues.adoc:70]: This is a really long sentence. I suggest breaking it up.

```suggestion
You must define either `velero` or `cloudstorage` when you configure a DPA. If you define both APIs in your DPA, the `openshift-adp-controller-manager` pod fails with a crash loop segmentation fault.
```

File: modules/oadp-features-plugins-known-issues.adoc
Comment (michaelryanpeter) [reply modules/oadp-features-plugins-known-issues.adoc:68]: Does this really need a second heading? It seems like the standard SSG format would be sufficient: https://redhat-documentation.github.io/supplementary-style-guide/#_known_issue

**Review by michaelryanpeter — COMMENTED** [review]
I hope my comments helped. Please reach out over Slack if you have any questions. Friendly reminder to get QE verification before putting this up for merge review.

/label peer-review-done
/remove-label peer-review-in-progress
/remove-label peer-review-needed

File: modules/oadp-features-plugins-known-issues.adoc
Comment (michaelryanpeter) [inline modules/oadp-features-plugins-known-issues.adoc:64]:
```suggestion
For more information about this issue, see link:https://issues.redhat.com/browse/OADP-1054[OADP-1054].
```
https://redhat-documentation.github.io/supplementary-style-guide/#cross-references

File: modules/oadp-features-plugins-known-issues.adoc
Comment (anarnold97) [reply modules/oadp-features-plugins-known-issues.adoc:59]: @michaelryanpeter - Velero supports backing up and restoring Kubernetes volumes using a free open-source backup tool called [restic](https://github.com/restic/restic). This support is considered beta quality. Please see the list of [limitations](https://velero.io/docs/v1.5/restic/#limitations) to understand if it fits your use case.

Velero allows you to take snapshots of persistent volumes as part of your backups if you’re using one of the supported cloud providers’ block storage offerings (Amazon EBS Volumes, Azure Managed Disks, Google Persistent Disks). It also provides a plugin model that enables anyone to implement additional object and block storage backends, outside the main Velero repository.

The restic integration was added to give you an out-of-the-box solution for backing up and restoring almost any type of Kubernetes volume. This integration is an addition to Velero’s capabilities, not a replacement for existing functionality. If you’re running on AWS, and taking EBS snapshots as part of your regular Velero backups, there’s no need to switch to using restic. However, if you need a volume snapshot plugin for your storage platform, or if you’re using EFS, AzureFile, NFS, emptyDir, local, or any other volume type that doesn’t have a native snapshot concept, restic might be for you.

Restic is not tied to a specific storage platform, which means that this integration also paves the way for future work to enable cross-volume-type data migrations.

File: modules/oadp-features-plugins-known-issues.adoc
Comment (anarnold97) [inline modules/oadp-features-plugins-known-issues.adoc:68]:
```suggestion
=== OpenShift ADP Controller segmentation fault workaround
```

File: modules/oadp-features-plugins-known-issues.adoc
Comment (bergerhoffer) [inline modules/oadp-features-plugins-known-issues.adoc:59]: "as" -> "because" per IBM style guide. But I wouldn't hold up the merge over just this.

image

**Review by bergerhoffer — COMMENTED** [review]
One minor grammar comment, but that's not critical.

However, I don't see QE approval on this PR. We require QE approval before merging PRs that have technical updates in them. 

I agree with Michael that I had some trouble trying to understand. Because line 57 mentions `cloudstorage` and `restic` (but not `velero`), but after that, only `cloudstorage` and `velero` are mentioned (but not `restic`).

And that's fine if we don't understand it, if it is actually technically correct, but that's what we need SME/QE approval to be sure of.  I checked the Jira and didn't see and approval there either, can you please send for QE review? It looks like there is a QE subtask on https://issues.redhat.com/browse/OADP-3716 that has a QE assigned.

File: modules/oadp-features-plugins-known-issues.adoc
Comment (anarnold97) [inline modules/oadp-features-plugins-known-issues.adoc:59]:
```suggestion
You can have either `velero` or `cloudstorage` defined, they are mutually exclusive fields.
```

**Comment by anarnold97** [comment]
Please see top of PR for approval from Wes

**Comment by bergerhoffer** [comment]
Hi @anarnold97, when I looked up Wes, he didn't appear to be a QE associate, so that wouldn't count. It has to be someone from the QE organization. I'm not familiar with the OADP but I'm assuming that the QA contact from the jira (@stillalearner) would be the right person to review. 

I'm removing the merge-review label for now, you can add it back in once you've gotten QE approval. Thanks!

File: modules/oadp-features-plugins-known-issues.adoc
Comment (bergerhoffer) [inline modules/oadp-features-plugins-known-issues.adoc:59]:
```suggestion
You can have either `velero` or `cloudstorage` defined, because they are mutually exclusive fields.
```

**Comment by anarnold97** [comment]
> Hi @anarnold97, when I looked up Wes, he didn't appear to be a QE associate, so that wouldn't count. It has to be someone from the QE organization. I'm not familiar with the OADP but I'm assuming that the QA contact from the jira (@stillalearner) would be the right person to review.
> 
> I'm removing the merge-review label for now, you can add it back in once you've gotten QE approval. Thanks!

You said "but that's what we need SME/QE approval to be sure of." Wes qualifies as an SME as lead dev on OADP.

File: modules/oadp-features-plugins-known-issues.adoc
Comment (anarnold97) [inline modules/oadp-features-plugins-known-issues.adoc:59]:
```suggestion
You can have either `velero` or `cloudstorage` defined, because they are mutually exclusive fields.
```

**Comment by anarnold97** [comment]
/label merge-review-neeed

The approval comes from the lead dev on OADP. Therefore qualifies as an SME.

**Comment by bergerhoffer** [comment]
@anarnold97 SME doesn't count, it must be from a QE associate. 

Please see our guidelines here on QE required before merging: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#verification-of-your-content

> All documentation changes that update or add technical content must be verified by a QE team associate before merging. 

And also the piece of the pull request template that states (in comments):

> QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. 

Please obtain QE approval before requesting a merge again. Let us know if you need help with requesting QE approval.

**Comment by anarnold97** [comment]
@bergerhoffer this was an urgent ticket for a customer
Yesterday was a holiday in India, and you have delayed this.
Wes is clearly more than qualified to know what changes in OADP are applicable.
I will delay this further by asking Sachin for a review


But I do not want you involved in any reviews for OADP documentation any more.

**Comment by anarnold97** [comment]
@stillalearner - please can you review

**Comment by kalexand-rh** [comment]
@anarnold97, all content additions require [QE review](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#verification-of-your-content), and Andrea's correct that this change does need QE. I think that any of the peer or merge reviewers would have either called out the lack of QE or review or tagged me in about it, so just asking Andrea to step back from OADP reviews won't "help."

**Review by stillalearner — APPROVED** [review]
lgtm

**Review by opayne1 — COMMENTED** [review]
lgtm

**Pull request merged**
